airbyte/progress.py

```diff
@@ -129,13 +140,38 @@ def __init__(
         self.last_update_time: float | None = None
 
         self._rich_view: RichLive | None = None
+
+        self.reset_progress_style(style)
+
+    def reset_progress_style(
+        self,
+        style: ProgressStyle = ProgressStyle.AUTO,
+    ) -> None:
+        """Set the progress bar style.
```

Comment:
@coderabbitai generate unit tests for this method. Especially important to cover mocks for scenarios that branch logic into different directions.

Reply:
@coderabbitai - Can you rewrite this using `pytest` instead of `unittest`?

Reply:
Incorporated here: https://github.com/airbytehq/PyAirbyte/pull/271/commits/f0b5c5b73462e8b31ad93b687b2f0e7332da0ac0